app/jobs/imo/invite_patient.rb

```diff
@@ -0,0 +1,29 @@
+class Imo::InvitePatient
+  include Sidekiq::Worker
+
+  sidekiq_options queue: :default
+
+  attr_reader :patient
+
+  def perform(patient_id)
+    @patient = Patient.find(patient_id)
+    result = client.invite
+    return if result == "failure"
```

Comment:
This should only happen due to poorly structured requests, so I don't think it should cause an infinite loop of invitations. These failures will be logged via sentry, so we should know if failures are happening frequently.

Reply:
If it does fail, it will go into the Sidekiq retry queue and retry some limited amount of times btw....it would not be infinite.  

That said, I don't think we should silently return if things fail.  I don't think we should ever get a generic `failure` if we handle things properly at the lower level.

I think these are the possible outcomes, lemme know what you think:

1. we send the invite and get back a valid response (`invited` or `no_imo_account`)
2. we send the invite and some sort of error happens at the API level ... the lower level job should raise and it will never get back here.  That may mean a timeout or some real error in IMO, or it may mean its some status comign back from IMO we need to add logic for to handle

If that is true, this all becomes

```
  patient = Patient.find(patient_id)
  result = ImpApiService.new(patient).invite
  ImoAuthorization.create!(patient: patient, status: result, last_invited_at: Time.current)
```

No need for instance state or a private method.

Reply:
Yeah, I didn't fully think through the fact that sidekiq needs the errors to cause retries. That's just an oversight on my part. I'll refactor.